src/software/simulated_tests/ball_moves_in_direction.py

```diff
@@ -34,8 +34,7 @@ def get_validation_status(self, world) -> ValidationStatus:
         # if max displacement is not set or current ball is moving in the right direction
         # set it and return PASSING
         if self.max_displacement_so_far is None or (
-            self.moving_in_pos_x
-            and current_ball_position > self.max_displacement_so_far - self.tolerance
+            current_ball_position > (self.max_displacement_so_far - self.tolerance)
```

Comment:
why did you remove the guard check for `self.moving_in_pos_x`

Reply:
Since that field didn't exist in this class. Not sure the error was merged. Also pretty sure that `ball_moves_forward.py` and `ball_moves_in_direction.py` are basically the same thing (cc: @sauravbanna)